app/src/test/java/org/mozilla/fenix/tabtray/DefaultTabTrayControllerTest.kt

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabtray
+
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import androidx.navigation.NavDirections
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import io.mockk.verify
+import io.mockk.verifyOrder
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.feature.tab.collections.TabCollection
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.components.TabCollectionStorage
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.sessionsOfType
+
+class DefaultTabTrayControllerTest {
+
+    private val activity: HomeActivity = mockk(relaxed = true)
+    private val navController: NavController = mockk(relaxed = true)
+    private val sessionManager: SessionManager = mockk(relaxed = true)
+    private val dismissTabTray: (() -> Unit) = mockk(relaxed = true)
+    private val showUndoSnackbar: ((String, SessionManager.Snapshot) -> Unit) =
+        mockk(relaxed = true)
+    private val registerCollectionStorageObserver: (() -> Unit) = mockk(relaxed = true)
+    private val tabCollectionStorage: TabCollectionStorage = mockk(relaxed = true)
+    val cachedTabCollections: List<TabCollection> = mockk(relaxed = true)
+    val currentDestination: NavDestination = mockk(relaxed = true)
```

Comment:
nit:
```suggestion
    private val currentDestination: NavDestination = mockk()
```

Since only the `id` getter is used and you mocked it, no need to relax the mock.